plugin.js

```diff
@@ -0,0 +1,92 @@
+const { spawn } = require('child_process');
+const { join, dirname } = require('path');
+const Watchpack = require('watchpack');
+const glob = require("glob");
+
+class WasmPackPlugin {
+  constructor(options) {
+    this.crateDirectory = options.crateDirectory;
+
+    this.wp = new Watchpack();
+  }
+
+  apply(compiler) {
+    this._compile(); // force first compilation
+
+    const files = glob.sync(join(this.crateDirectory, '**', '*.rs'));
+
+    this.wp.watch(files, [], Date.now() - 10000);
+    this.wp.on('change', this._compile.bind(this));
+  }
+
+  _compile() {
+    console.log('ℹ️  Compiling your crate...\n');
+
+    spawnWasmPack({
+      isDebug: true,
+      cwd: this.crateDirectory
+    })
+      .then(this._compilationSuccess)
+      .catch(this._compilationFailure);
+  }
+
+  _compilationSuccess(info) {
+    if (info) {
+      console.log(info);
+    }
+
+    console.log('✅  Your crate has been correctly compiled\n');
+  }
+
+  _compilationFailure(err) {
+    throw err;
+  }
+}
+
+function spawnWasmPack({ isDebug, cwd }) {
+  const bin = 'wasm-pack';
+
+  const args = [
+    '--verbose',
```

Comment:
WDYT about using things like `Commander` here?

Reply:
You mean https://www.npmjs.com/package/commander? Not sure you can spawn a program with that? But it seems to be trivial enough.